internal/ingress/metric/collectors/socket.go

```diff
@@ -119,6 +122,8 @@ func NewSocketCollector(pod, namespace, class string) (*SocketCollector, error)
 	sc := &SocketCollector{
 		listener: listener,
 
+		workerPool: pool.NewLimited(1),
```

Comment:
Would the lock issue still persist if you use more than one worker?

My understanding is there’s not necessarily a dead lock - we should still be able to parallelize this.

Would be great to test and confirm this though.

With single worker when there are many nginx workers we might end up dropping some metrics.

Reply:
> Would the lock issue still persist if you use more than one worker?

This ensures we process only one batch at the time, no matter how many workers are configured

> My understanding is there’s not necessarily a dead lock - we should still be able to parallelize this.

That's correct but if we parallelize the update of the metrics we will hit the same lock issue in prometheus

> With single worker when there are many nginx workers we might end up dropping some metrics

Why you reach to this conclusion, the pool ensures we only process one batch at the time but we enqueue ALL the requests received by the lua request

Reply:
> That's correct but if we parallelize the update of the metrics we will hit the same lock issue in prometheus

I had an impression that locking becomes an issue when there are many Goroutines, but having few fixed number of Goroutines would be helpful.

> the pool ensures we only process one batch at the time but we enqueue ALL the requests received by the lua request

👍